Chaincase.Common/Global.cs

```diff
@@ -480,6 +488,20 @@ public async Task OnSleeping()
                     }
 
                     IsGoingToSleep = false;
+
+                    if (_P2EPServer.HiddenServiceIsOn)
+                    {
+                        //KeepHiddenServiceAlive.Dispose();
+                        //KeepHiddenServiceAlive = new CancellationTokenSource();
+                        //// wait 20 seconds
+                        //for (int i = 0; i < 20; i++)
+                        //{
+                        //    await Task.Delay(1_000); //
+                        //    if (KeepHiddenServiceAlive.IsCancellationRequested || !P2EPServer.HiddenServiceIsOn) return;
+                        //}
+                        await _P2EPServer.StopAsync(SleepCts.Token);
```

Comment:
same comment as above. P2EPServer's lifecycle ought to be self contained and non reliant on external control conditions